lib/saml_idp/service_provider.rb

```diff
@@ -22,18 +22,13 @@ def valid?
     end
 
     def valid_signature?(doc, require_signature = false)
-      if require_signature || should_validate_signature?
+      if require_signature || attributes[:validate_signature]
         doc.valid_signature?(fingerprint)
       else
         true
       end
     end
 
-    def should_validate_signature?
```

Comment:
Interesting. This would also negate the need for `SamlIdp::PersistedMetadata` since it only implements `sign_assertions?` which is only used here. 

I'm not sure I fully grok the last part:
> This requirement is in addition to any requirement for signing derived from the use of a particular profile/binding combination.

Reply:
@johnf  to be honest, I'm also not 100% sure what it's exactly mean, since they wrote their docs like puzzles. As I far understand they are mentioning that others 2 specification such as profile/binding should also see for this value. Because those parts also have `<saml:Assertion>`.
 
https://docs.oasis-open.org/security/saml/v2.0/saml-profiles-2.0-os.pdf, line 634
https://docs.oasis-open.org/security/saml/v2.0/saml-bindings-2.0-os.pdf, line 425

Maybe I'm missing some part also.

Reply:
I think I see now. That Metadata code is mostly a bunch of lying code. Looking at one's own old code is tough some times, since I now see a lot of problems 😂 Oh well, I will have to document some rework.

I think this code is good, though. Will approve and we can merge shortly.